meerkat_abacus/util/__init__.py

```diff
@@ -76,21 +79,69 @@ def get_db_engine(db_url=config.DATABASE_URL):
     return engine, session
 
 
-def epi_week_start_date(year, epi_config=country_config["epi_week"]):
+def epi_year_start_date(date, epi_config=country_config["epi_week"]):
     """
-    Get the first day of epi week 1
+    Get the first day of epi week 1 for year including given date.
 
     if epi_config==international epi_week 1 starts on the 1st of January
 
-    if epi_config== day:X then the first epi_week start on the first weekday
-    X after 1st of January
-    X=0 is Sunday
+    if epi_config== day:<weekday> then the first epi_week start on the first weekday
+    First weekday after 1st of January
+    <weekday> is an integer where <weekday>=0 is Monday, 2 is Tuesday, etc.
+
+    if epi_config is a dict one can specify custom start dates for epi years
+    e.g.
+    "epi_week": {
+        2011: datetime.datetime(2011, 1, 2),
+        2012: datetime.datetime(2011, 12, 31)
+    }
 
     Args:
-        year: year
+        date: date for which to return the start of epi year
+        epi_config: how epi-weeks are calculated
+    Returns:
+        start_date: date of start of epi week 1 which includes provided date
+    """
+    if epi_config == "international":
+        return datetime(date.year, 1, 1)
+    elif "day" in epi_config:
+        day_of_week = int(epi_config.split(":")[1])
+        first_of_year = datetime(date.year, 1, 1)
+        f_day_of_week = first_of_year.weekday()
+        adjustment = day_of_week - f_day_of_week
+        if adjustment < 0:
+            adjustment = 7 + adjustment
+        return first_of_year + timedelta(days=adjustment)
+    elif isinstance(epi_config, dict):
+        for start_datetime in reversed(sorted(epi_config.values())):
```

Comment:
Not sure if this is much better but maybe we can replace reversed(.....) with something like this: 
[ epi_config[year] for year in [year - 1, year, year +1]]
Where year is the year of the current date. We should only have to check these three values I think.

Reply:
I don't think that performance is an issue here. Sorting and reversing a collection of < 50 elements should be very fast. IMO current solution is clearer and simpler.

Reply:
Fair enough.